metricbeat/module/kibana/cluster_actions/data.go

```diff
@@ -0,0 +1,123 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cluster_actions
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/elastic/beats/v7/metricbeat/helper/elastic"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	s "github.com/elastic/beats/v7/libbeat/common/schema"
+	c "github.com/elastic/beats/v7/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+var (
+	kibanaSchema = s.Schema{
+		"uuid":  c.Str("uuid"),
+		"name":  c.Str("name"),
+		"index": c.Str("index"),
+		"host": s.Object{
+			"name": c.Str("host"),
+		},
+		"transport_address": c.Str("transport_address"),
+		"version":           c.Str("version"),
+		"snapshot":          c.Bool("snapshot"),
+		"status":            c.Str("status"),
+	}
+
+	actionsSchema = s.Schema{
+		"overdue": c.Dict("overdue", s.Schema{
+			"count": c.Int("count"),
+			"delay": c.Dict("delay", s.Schema{
+				"p50": c.Float("p50"),
+				"p99": c.Float("p99"),
+			}),
+		}),
+	}
+)
+
+type response struct {
+	Actions     map[string]interface{} `json:"cluster_actions"`
+	Kibana      map[string]interface{} `json:"kibana"`
+	ClusterUuid string                 `json:"cluster_uuid"`
+}
+
+func eventMapping(r mb.ReporterV2, content []byte, isXpack bool) error {
+	var data response
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		return fmt.Errorf("failure parsing Kibana Cluster Actions API response: %w", err)
+	}
+
+	rootFields := mapstr.M{}
+	moduleFields := mapstr.M{}
+
+	// Set elasticsearch cluster id
+	//nolint: errcheck // This should not fail
+	moduleFields.Put("elasticsearch.cluster.id", data.ClusterUuid)
+
+	kibana, _ := kibanaSchema.Apply(data.Kibana)
+	if err != nil {
+		return elastic.MakeErrorForMissingField("kibana", elastic.Kibana)
+	}
+
+	// Set service ID
+	serviceId, err := kibana.GetValue("uuid")
+	if err != nil {
+		return elastic.MakeErrorForMissingField("kibana.uuid", elastic.Kibana)
+	}
+	//nolint: errcheck // This should not fail
+	rootFields.Put("service.id", serviceId)
```

Comment:
We can avoid all these put statements (and resulting `nolint` statements) if we construct the map after we fetch all the data, like this: https://github.com/elastic/beats/blob/84fc06b99e32e505076275ad41a6c1b204bd9fc7/metricbeat/module/system/diskio/diskio.go#L87

`mapstr.Put()` is good for conditional logic, when we want to add something inside an `if` block, but if we have all the fields, we're probably better off constructing the map all at once.